src/material_library.cpp

```diff
@@ -121,9 +121,8 @@ void pyne::MaterialLibrary::add_material(pyne::Material mat) {
   if (mat.metadata.isMember("mat_number")) {
     mat_number = mat.metadata["mat_number"].asInt();
     mat_numb_it = mat_number_set.find(mat_number);
-    if (mat_numb_it == mat_number_set.end()) {
-      warning("The Material Number Conflict, the material has not been added to the database.");
-      return;
+    if (mat_numb_it != mat_number_set.end()) {
+      warning("The Material Number Conflict.");
```

Comment:
Suggestion: print this `mat_numb_it` in the warning message.

Reply:
done thx for the suggestion!